bin/conjur-cli.rb

```diff
@@ -36,12 +36,24 @@ def test_select
   true
 end
 
+def file_input(path)
+  if path == '-'
+    $stdin.read.force_encoding('ASCII-8BIT')
+  else
+    File.read(path, mode: 'rb')
+  end
+end
```

Comment:
It looks like there are a number of potential failure cases here. Do these manifest in the messages you would want the user to see?

For example:

- The file doesn't exist
- The file is empty
- The `-` argument is given, but nothing is provided to STDIN

Along these lines, is there any way some of this could be unit tested? Or is there no infrastructure/fixture for that now with this CLI?

Reply:
If the file doesn't exist the user will see the following error:
`conjurctl server --account demo --password-file /no_such_file
error: No such file or directory @ rb_sysopen - /no_such_file`. 
I can wrap this error if you think it would provide a better UX.

If the `-` argument is given, but nothing is provided to STDIN ex. 
`conjurctl server --account demo --password-file`, it will hang. 
However, a user could still provide input by typing in a password and then Ctrl+D.
I could add a timeout if you think this would improve the UX, but there are pros and cons to that approach as well. 

We currently do not have any unit tests for conjurctl. I will look into adding a cucumber test for `conjurctl account create`

Reply:
Regarding:
>If the - argument is given, but nothing is provided to STDIN ex.
conjurctl server --account demo --password-file, it will hang.
However, a user could still provide input by typing in a password and then Ctrl+D.
I could add a timeout if you think this would improve the UX, but there are pros and cons to that approach as well.

You can actually detect this without a timeout using `$stdin.tty?`
| command | `$stdin.tty?` |
| - | - |
| <code>echo 'password' &#124; conjurctl ... --password-file -</code> | false |
| `conjurctl ... --password-file - < password-file.txt` | false |
| `conjurctl ... --password-file -` | true |

That would give you either an opportunity to interactively prompt for it, or just raise an error if the standard input isn't piped int.

Reply:
TIL